line_beam_parameters.py

```diff
@@ -27,21 +29,38 @@ class LineBeamParametersControl(Display):
     _setting_bits = False
     energy_channel = None
 
+    # Monitor the rate readback for use in the zero rate button
+    rate_channel = None
+
+    # Signal to set a new rate from the combobox or from zero rate button
+    update_rate_signal = QtCore.Signal(int)
+
+
     def __init__(self, parent=None, args=None, macros=None):
         super(LineBeamParametersControl, self).__init__(parent=parent,
                                                         args=args,
                                                         macros=macros)
         self.config = macros
         self.setup_ui()
 
-        if self.energy_channel:
+        if self.energy_channel is not None:
             self.destroyed.connect(functools.partial(clear_channel,
```

Comment:
Just a thought, but this seems like a useful pattern to tear out (take in or create channel, clear it up when widget x is destroyed)

Reply:
I think you're right. I think PyDM does some search through all the "channels" attributes (or some similarly named attribute) at shutdown to clear all the channels, but not all widgets live until the close of the application. Something with a signature like:
```
def add_channel(widget: QWidget, <pydm channel args>) -> PyDMChannel:
    ...
```
That did the common setup/teardown legwork would be useful.
- create a PyDMChannel
- start the channel connection in parallel (or in series if needed)
- set up the destroyed signal

Reply:
Actually, slightly cleaner:
```
def add_channel(widget: QWidget, channel: PyDMChannel) -> PyDMChannel:
    ...
```

Reply:
I like it 👍